File: AndroidDemo/app/src/main/java/com/ziwenl/androiddemo/router/RouterPath.kt
package com.ziwenl.androiddemo.router

/**
 * PackageName : com.ziwenl.androiddemo.router
 * Author : Ziwen Lan
 * Date : 2021/1/20
 * Time : 10:57
 * Introduction : Flutter / Native 页面路由地址
 * 路由地址应与 Flutter 模块中设定的一致
 * 按模块划分
 */
class RouterPath {
    companion object {
        const val NATIVE = "native"
        const val FLUTTER = "flutter"

        //路由协议
        /**
         * 指向 Native 页面
         */
        const val PATH_TYPE_NATIVE = "$NATIVE://";

        /**
         * 指向 Flutter 页面
         */
        const val PATH_TYPE_FLUTTER = "$FLUTTER://";
    }

    /**
     * 测试用页面
     */
    object Test {
        private const val TEST_NATIVE = "${PATH_TYPE_NATIVE}/test"
        private const val TEST_FLUTTER = "${PATH_TYPE_FLUTTER}/test"

        const val PAGE_HAS_RESULT = "$TEST_FLUTTER/has_result"

        const val PAGE_HAS_PARAMS = "$TEST_FLUTTER/has_params"

        /**
         * 带回调的 native 页面
         */
        const val PAGE_NATIVE_HAS_RESULT = "$TEST_NATIVE/has_result"
    }

    /**
     * 基础页面
     */
    object Base {
        private const val TEST_NATIVE = "${PATH_TYPE_NATIVE}/test"
        private const val TEST_FLUTTER = "${PATH_TYPE_FLUTTER}/test"

    }

    /**
     * 主页面
     */
    object Main {
        private const val MAIN_NATIVE = "${PATH_TYPE_NATIVE}/main"
        private const val MAIN_FLUTTER = "${PATH_TYPE_FLUTTER}/main"
        const val PAGE_MAIN = "$MAIN_FLUTTER/main"
    }

    /**
     * 首页
     */
    object Home {
        private const val HOME_NATIVE = "${PATH_TYPE_NATIVE}/home"
        private const val HOME_FLUTTER = "${PATH_TYPE_FLUTTER}/home"

    }

    /**
     * 我的
     */
    object Mine {
        private const val MINE_NATIVE = "${PATH_TYPE_NATIVE}/mine"
        private const val MINE_FLUTTER = "${PATH_TYPE_FLUTTER}/mine"

    }
}

File: flutter_module/lib/router/RouterPath.dart
//路由协议
class RouterPath {
  //指向 Native 页面
  static const PATH_TYPE_NATIVE = "native://";
  //指向 Flutter 页面
  static const PATH_TYPE_FLUTTER = "flutter://";
}

// 测试用页面
class Test {
  static const _TEST_NATIVE = "${RouterPath.PATH_TYPE_NATIVE}/test";
  static const _TEST_FLUTTER = "${RouterPath.PATH_TYPE_FLUTTER}/test";

  static const String PAGE_HAS_RESULT = "$_TEST_FLUTTER/has_result";
  static const String PAGE_HAS_PARAMS = "$_TEST_FLUTTER/has_params";

  // 带回调的 native 页面
  static const PAGE_NATIVE_HAS_RESULT = "$_TEST_NATIVE/has_result";
}

// 基础页面
class Base {
  static const _BASE_NATIVE = "${RouterPath.PATH_TYPE_NATIVE}/base";
  static const _BASE_FLUTTER = "${RouterPath.PATH_TYPE_FLUTTER}/base";

  static const String PAGE_NORMAL = "$_BASE_FLUTTER/normal";
}

// 主页面
class Main {
  static const _MAIN_NATIVE = "${RouterPath.PATH_TYPE_NATIVE}/main";
  static const _MAIN_FLUTTER = "${RouterPath.PATH_TYPE_FLUTTER}/main";

  static const String PAGE_MAIN = "$_MAIN_FLUTTER/main";
}

// 首页
class Home {
  static const _HOME_NATIVE = "${RouterPath.PATH_TYPE_NATIVE}/home";
  static const _HOME_FLUTTER = "${RouterPath.PATH_TYPE_FLUTTER}/home";

  static const String PAGE_HOME = "$_HOME_FLUTTER/home";
}

// 我的
class Mine {
  static const _MINE_NATIVE = "${RouterPath.PATH_TYPE_NATIVE}/mine";
  static const _MINE_FLUTTER = "${RouterPath.PATH_TYPE_FLUTTER}/mine";

  static const String PAGE_MINE = "$_MINE_FLUTTER/mine";
}
